_posts/2019-03-06-generating-print-quality-plots.md

```diff
@@ -0,0 +1,128 @@
+
+The equivalent way to save to PNG file via the code is this:
+```R
+png('heatMap.png', pointsize=10, width=700, height=480)
+filled.contour(lons,lats,area,
+                   xlim = range(lons), ylim = range(lats),
+                   plot.axes = {axis(1);axis(2);
+                     map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+dev.off()
+```
+
+Increasing the resolution can be a bit tricky.
+```R
+png('heatMap2.png', pointsize=10, width=700, height=480, res=300)
+filled.contour(lons,lats,area,
+                   xlim = range(lons), ylim = range(lats),
+                   plot.axes = {axis(1);axis(2);
+                     map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+dev.off()
+```
+![Changed resolution to 300dpm]({{ "/images/heatmap2.PNG" | absolute_url }})
+Adding the `res` parameter fixes up the resolution, but it also rescales the axes and text. Therefore, you need to increase the height and the width.
+
+```R
+png('heatmap3.png', pointsize=10, width=1400, height=960, res=300)
+filled.contour(lons,lats,area,
+               xlim = range(lons), ylim = range(lats),
+               plot.axes = {axis(1);axis(2);
+                 map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+dev.off()
+```
+![Doubled width and height]({{ "/images/heatmap3.PNG" | absolute_url }})
+
+That's better.
+
+What if we want an even higher resolution, say 600dpi. So we change the first line to:
+```R
+png('C:/Users/Danny/Documents/heatmap4.png', pointsize=10, width=2800, height=2000, res=600)
+```
+and run the plot again.
+This time you might get the following error: `Error in plot.new() : figure region too large`.
+
+What this means is that your figure doesn't fit in the default device screen that R has set. What does that mean? Well, when you run a `png()` command, you are actually telling R to set up a PNG output device on which to display your plot. That device has certain default parameters, including dimensions and margin sizes. In this case, the plot you have specified doesn't fit in the device's spatial parameters.
+
+You may be able to solve this problem by changing the device's margin sizes. This may require a bit of fiddling to get them right. Margin sizes are measured in units of lines of text. (A plot has a specific text size.)
+
+By adding a `par(mar=...)` command, my error went away.
+```R
+png('C:/Users/Danny/Documents/heatmap5.png', pointsize=10, width=2800, height=2000, res=600)
+par(mar=c(5, 4, 4, 2))
+filled.contour(lons,lats,area,
+               xlim = range(lons), ylim = range(lats),
+               plot.axes = {axis(1);axis(2);
+                 map('world',add=TRUE, wrap=c(-180,180), interior = FALSE)})
+dev.off()
+```
+
+You may also get an error message saying that the margins are too big. That might indicate that:
+1. You have to make them smaller, or
+2. You haven't set your plot dimensions to be sufficiently large, so there isn't enough space to fit the margins.
```

Comment:
Considering the issues with increasing the resolution of `png` format, might it also be a good idea to show producing `pdf` using `dev` and then explaining there are post-processing tools that can convert a `pdf` to `png`? On the mac Preview does it pretty handily. There is also the cross platform ImageMagick suite, or people may have the Adobe Acrobat suite installed on their windows machines. I don't think there needs to be a lot of detail. People can google it themselves if they really need to.